model/map/src/main/java/org/keycloak/models/map/serverinfo/MapServerInfoProviderFactory.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.models.map.serverinfo;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import org.jboss.logging.Logger;
+import org.keycloak.Config;
+import org.keycloak.common.Version;
+import org.keycloak.common.util.Base64Url;
+import org.keycloak.common.util.RandomString;
+import org.keycloak.migration.MigrationModel;
+import org.keycloak.migration.ModelVersion;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.ServerInfoProvider;
+import org.keycloak.models.ServerInfoProviderFactory;
+import org.keycloak.models.map.common.AbstractMapProviderFactory;
+
+public class MapServerInfoProviderFactory extends AbstractMapProviderFactory<ServerInfoProvider> implements ServerInfoProviderFactory {
+
+    private static final String RESOURCES_VERSION_SEED = "resourcesVersionSeed";
```

Comment:
What is the purpose of this seed?

Reply:
It is useful in clustered scenarios. When this seed is not set in config, the random seed is generated and used. In clustered environment it might be useful to set the seed yourself for having the same `Version.RESOURCES_VERSION` across all nodes in cluster. Then all resources (js, css, ...) could have the same url.

Reply:
I see. Ok thanks for the explanation.